zerver/lib/markdown/fenced_code.py

```diff
@@ -392,6 +393,13 @@ def format_code(self, lang: str, text: str) -> str:
         else:
             code = CODE_WRAP.format(langclass, self._escape(text))
 
+        # In order to display a "view-in-playground" option in the frontend,
+        # we need to know the language used in the codeblock. We tweak the HTML
+        # CodeHilite generates to add this language as a data-attribute.
+        if lang:
+            parsed_code = BeautifulSoup(code, features='lxml')
```

Comment:
Do we need BeautifulSoup here?  I'd have expected that we'd be able to use LXML directly without a `find` step, since the outer element of `code` is known to be the `div`.